bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -1323,14 +1341,14 @@ private OperationType getOperationType(byte opCode) {
     private void readV3Response(final Response response) {
         final BKPacketHeader header = response.getHeader();
 
-        final CompletionValue completionValue = completionObjects.remove(newCompletionKey(header.getTxnId(),
+        final CompletionValue completionValue = completionObjects.get(newCompletionKey(header.getTxnId(),
```

Comment:
any reason why change remove to get?

if seems a new completion key is created for each get and remove, at least if we can use same key that would reduce garbages.